## Assignment No.6

# Hypothesis Testing

In [15]:
import scipy.stats as stats
import numpy as np
x_bar   = 3050     
X_mean  = 600  
sigma_X = 25       
n       = 25      
alpha   = 0.05  

In [16]:
print("H₀: μ = Rs. 4,000  (costs are as per model)")
print("H₁: μ > Rs. 4,000  (costs are higher – owners' claim)")

H₀: μ = Rs. 4,000  (costs are as per model)
H₁: μ > Rs. 4,000  (costs are higher – owners' claim)


In [17]:
# Right-tailed test at α = 0.05
z_critical = stats.norm.ppf(1 - alpha)
p_value    = 1 - stats.norm.cdf(z)     
print(f"Critical Value (z)  = {z_critical:.4f}")
print(f"p-value             = {p_value:.6f}")
print(f"Alpha (α)           = {alpha}")


Critical Value (z)  = 1.6449
p-value             = 1.000000
Alpha (α)           = 0.05


In [18]:
# Right-tailed test at α = 0.05
z_critical = stats.norm.ppf(1 - alpha) 
p_value    = 1 - stats.norm.cdf(z)  

print(f"Critical Value (z)  = {z_critical:.4f}")
print(f"p-value             = {p_value:.6f}")
print(f"Alpha (α)           = {alpha}")


Critical Value (z)  = 1.6449
p-value             = 1.000000
Alpha (α)           = 0.05


CONCLUSION:
The calculated test statistic z = -38.0 is far LESS than the critical value of +1.6449.
Also, p-value = 1.0 which is much GREATER than α = 0.05.
Therefore, we FAIL TO REJECT the Null Hypothesis (H₀).

This means there is NO sufficient statistical evidence to support the restaurant owners' claim that their weekly operating costs are HIGHER than what the model predicts.
(W = Rs. 4,000 for 600 units).
In fact, the sample mean (Rs. 3,050) is LOWER than the theoretical mean (Rs. 4,000), which contradicts the owners' claim entirely.

## Estimation And Confidence Intervals

In [20]:
import numpy as np
from scipy import stats

data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)
x_bar = np.mean(data)
s = np.std(data, ddof=1)
se = s / np.sqrt(n)

alpha = 0.01
df = n - 1
t_critical = stats.t.ppf(1 - alpha/2, df)
margin_t = t_critical * se
ci_t_lower = x_bar - margin_t
ci_t_upper = x_bar + margin_t

print("===== Part A: 99% CI Using Sample Standard Deviation (t-distribution) =====")
print(f"Sample Size (n)          : {n}")
print(f"Sample Mean (x̄)          : {x_bar:.4f}")
print(f"Sample Std Dev (s)       : {s:.4f}")
print(f"Standard Error (SE)      : {se:.4f}")
print(f"Degrees of Freedom (df)  : {df}")
print(f"t-Critical Value         : {t_critical:.4f}")
print(f"Margin of Error          : {margin_t:.4f}")
print(f"99% Confidence Interval  : ({ci_t_lower:.4f}, {ci_t_upper:.4f})")

===== Part A: 99% CI Using Sample Standard Deviation (t-distribution) =====
Sample Size (n)          : 15
Sample Mean (x̄)          : 1.2387
Sample Std Dev (s)       : 0.1932
Standard Error (SE)      : 0.0499
Degrees of Freedom (df)  : 14
t-Critical Value         : 2.9768
Margin of Error          : 0.1485
99% Confidence Interval  : (1.0902, 1.3871)


In [21]:
sigma = 0.2
se_z = sigma / np.sqrt(n)
z_critical = stats.norm.ppf(1 - alpha/2)
margin_z = z_critical * se_z
ci_z_lower = x_bar - margin_z
ci_z_upper = x_bar + margin_z

print("===== Part B: 99% CI Using Known Population Standard Deviation (z-distribution) =====")
print(f"Population Std Dev (σ)   : {sigma}")
print(f"Standard Error (SE)      : {se_z:.4f}")
print(f"z-Critical Value         : {z_critical:.4f}")
print(f"Margin of Error          : {margin_z:.4f}")
print(f"99% Confidence Interval  : ({ci_z_lower:.4f}, {ci_z_upper:.4f})")

===== Part B: 99% CI Using Known Population Standard Deviation (z-distribution) =====
Population Std Dev (σ)   : 0.2
Standard Error (SE)      : 0.0516
z-Critical Value         : 2.5758
Margin of Error          : 0.1330
99% Confidence Interval  : (1.1057, 1.3717)
